fix: Show recordings for interims

richsalz opened the pull request:
List the recordings if the "meeting number" starts with "interim"

Fixes: #6543

richsalz committed:
fix: Show recordings for interims

Add methods uses_notes(), has_recordings(), and uses_chat_logs() to the
meeting object (with semantically correct tests) and use them consistently
throughout.  List the recordings if the "meeting numnber" starts with
"interim"

Fixes: #6543

richsalz commented:
Should the same fix be made for the buttons, at https://github.com/ietf-tools/datatracker/blob/main/ietf/templates/meeting/session_buttons_include.html#L139 ?

rjsparks commented on ietf/templates/meeting/session_details_panel.html:
```diff
@@ -320,7 +320,7 @@ <h3 class="mt-4">Notes and recordings</h3>
             {% endif %}
             {# Recordings #}
-            {% if meeting.number|add:"0" >= 80 %}
+            {% if meeting.number|add:"0" >= 80  or "interim" in meeting.number %}
```
```suggestion
            {% if meeting.type_id == 'ietf' and meeting.number|add:"0" >= 80  or meeting.type_id != 'ietf' %}
```

rjsparks replied:
The above suggestion is a bit awkward - it requires knowing the precedence of operators as documented at https://docs.djangoproject.com/en/5.0/ref/templates/builtins/#boolean-operators. It might be better to move this into the view code to get away from the `|add:"0"` type-coersion-by-hack-trick anyhow.

richsalz replied:
I thought of something like that.  In addition to the one mentioned above, the other [tests against meeting 60](https://github.com/ietf-tools/datatracker/blob/main/ietf/templates/meeting/session_buttons_include.html#L129).  What's the significance of meetings 60 and 80?  I'd like to remove the magic numbers, replacing with a reasonable function name.

rjsparks replied:
they aren't magic (though they aren't complete) - they are places where the _shape_ of proceedings and what was captured at meetings fundamentally changes.  60 was when we first started capturing chatlogs (jabber at the time). 80 was a point at where recording tech changed significantly.

That said, we don't have any Documents of type "recording" before ietf-100, so the code being touched here could could (for meetings of type ietf) not show the recordings block at all for meeting numbers < 100.

rjsparks commented:
>Should the same fix be made for the buttons, at https://github.com/ietf-tools/datatracker/blob/main/ietf/templates/meeting/session_buttons_include.html#L139 ?

Looks like it.

This really needs a new (or updated) test.

rjsparks commented on ietf/doc/templatetags/ietf_filters.py:
```diff
@@ -881,3 +881,19 @@ def badgeify(blob):
     return text
+
+@register.filter
+def has_chat_logs(meeting):
```
Should this get the same "Return true for interims" treatment?

rjsparks replied:
It's not clear that these should be filters - why aren't they methods on Meeting instead? (those would be visible to the templates). I think you were on the way to discovering that with the partial removal of the has_notes filter, and the redundant argument passed to the templates in the views later.

richsalz replied:
Happy to make them methods on the meeting if you prefer. I didn't know templates could invoke methods directly :)

richsalz replied:
> Should this get the same "Return true for interims" treatment?

I guess?  I changed it in the commit 466a6306a

rjsparks commented on ietf/doc/templatetags/ietf_filters.py:
```diff
@@ -881,3 +881,19 @@ def badgeify(blob):
+    num = meeting.get_number()
+    return num != None and num >= 80
+
+#register.filter
+#define has_notes(meeting):
+    return meeting.uses_notes()
```
This looks incomplete - if the intent was to remove the filter L899 should also go.

rjsparks replied:
(and I suspect all of these should just be removed once the functionality is moved to Meeting.

richsalz replied:
yes, this was just checking in a snapshot before i got on the plane.  will move to templates and fix shortly.

rjsparks reviewed (changes requested):
This is shaping up nicely. It should go a litter further as noted inline.

richsalz commented:
Rebased, squashed, fixed the tests. This is ready for review now.

jennifer-richards commented on ietf/meeting/views.py:
```diff
@@ -1691,7 +1690,7 @@ def api_get_agenda_data (request, num=None):
         "categories": filter_organizer.get_filter_categories(),
         "isCurrentMeeting": is_current_meeting,
-        "useNotes": meeting.uses_notes(),
+        "uses_notes": meeting.uses_notes(),
```
Should also stay as a camelCase key (`usesNotes`) to match the rest of the structure

jennifer-richards reviewed (changes requested):
Overall looks good - thank you for getting rid of the kludgy template code! There are a few nits to adjust that I've noted inline but I don't see any big issues.

richsalz committed:
style: Use "is not" and "is" for None comparisons

richsalz committed:
None comparison and non-IETF meetings

style: Use "is not None" instead of "!="
For non-IETF meetings assume chat logs exist

richsalz committed:
fix: Restore useNotes for JS fields

richsalz commented:
@jennifer-richards I believe I addressed all your feedback, PTAL.

jennifer-richards commented on ietf/meeting/models.py:
```diff
@@ -385,17 +385,19 @@ def uses_notes(self):
 
     def has_chat_logs(self):
+        if self.type_id != 'ietf':
+            return True;
```
I think this should be `False` to preserve the behavior? For a non-ietf meeting, `self.get_number()` returns `None`, so it used to return `False`. (If it should be `True` or is a don't care, just say so, I'm just pointing out the change.)

rjsparks replied:
I think what's here now matches current and desired behavior. What existed earlier in this branch may have been off base about whether interims had chat logs, but the do (and always have).

jennifer-richards reviewed (commented):
One question inline

rjsparks reviewed (changes requested):
Let me know if I misread the request for keeping the js variables in camelCase.

richsalz committed:
fix: uses_notes->useNotes (in JavaScript)

Also add comment about meeting number field in tests

richsalz commented on ietf/meeting/tests_views.py:
```diff
@@ -259,7 +259,7 @@ def test_meeting_agenda(self):
                 "categories": rjson.get("categories"), # Just expect the value to exist
                 "isCurrentMeeting": True,
-                "useNotes": True,
+                "usesNotes": False,
```
I added the archeological comment.  And fixed the other JS case.

jennifer-richards commented on client/agenda/AgendaScheduleList.vue:
```diff
@@ -296,7 +296,7 @@ const meetingEvents = computed(() => {
         })
       }
-      if (agendaStore.useNotes) {
+      if (agendaStore.uses_notes) {
```
I think this one needs `usesNotes` also?

richsalz replied:
dang.  but no change in test result, so I missed it.  fix pushed.

jennifer-richards reviewed (changes requested):
I think there's still one variable name change needed

richsalz committed:
Missed a uses_notes->useNotes edit

jennifer-richards committed:
fix: useNotes->usesNotes

jennifer-richards commented:
I think there was still disagreement about `useNotes` vs `usesNotes`. Easier to push a commit than to point at them, but please review that I'm not mixing up similarly named variables in 4e8ce54cdbaeb35c77c370ba86e90a29ab629376

rjsparks committed:
Merge remote-tracking branch 'ietf-tools/main' into fix-6543

TohaAbir commented:
I need some time. I lost my phone. I am need to recover information how
check my activity

On Wed, Aug 7, 2024, 9:43 PM Robert Sparks ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ietf-tools/datatracker/pull/7197#pullrequestreview-2225571227>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJFWHDWYP25WRF6ERBDYFELZQI6B5AVCNFSM6AAAAABEZAUU42VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDEMRVGU3TCMRSG4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
